### PR TITLE
chore(jest): transformIgnorePatterns break on Windows or with pnpm

### PR DESCRIPTION
**Pull request by KotoriK**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Workflow
- [ ] Other (about what?)

#### background
jest reports token not supports, which is caused by loading ES Module unintently.
#### solution
change ```transformIgnorePatterns```.
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Jest config which used to fail to support Windows and pnpm         |
| 🇨🇳 Chinese | 修复Jest配置不支持Windows或使用pnpm的开发环境的问题  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at df3bae0</samp>

Improve test configuration in `.jest.js` to support pnpm and Windows. Adjust `compileModules` and `transformIgnorePatterns` to handle different module paths and separators.

### <samp>🤖 Generated by Copilot at df3bae0</samp>

* Enable tests to run with pnpm by adding `.pnpm` folder to `compileModules` array ([link](https://github.com/ant-design/ant-design/pull/45557/files?diff=unified&w=0#diff-6dc98a965c38d41d2ecabb09c6a6842bef943e69bcd6da63d9371efa466f2ca4L1-R8))
* Fix tests failing on Windows by using platform-independent path separators in `transformIgnorePatterns` regex ([link](https://github.com/ant-design/ant-design/pull/45557/files?diff=unified&w=0#diff-6dc98a965c38d41d2ecabb09c6a6842bef943e69bcd6da63d9371efa466f2ca4L15-R22))

### CODE REVIEW

File: .jest.js [hunk 1]
```diff
@@ -12,7 +19,7 @@ const ignoreList = [];
 const transformIgnorePatterns = [
   // Ignore modules without es dir.
   // Update: @babel/runtime should also be transformed
-  `/node_modules/(?!${ignoreList.join('|')})[^/]+?/(?!(es)/)`,
+  `[\\/]node_modules[\\/](?!${ignoreList.join('|')})[^\\/]+?[\\/](?!(es)[\\/])`,
```
Comment (yoyo837) [inline .jest.js:22]: 这样写了非Windows的不就挂了?

Comment (KotoriK) [reply .jest.js:22]: 额 或许你可以去[regexr.com](https://regexr.com)跑个测试看看
完整表达式:```[\\/]node_modules[\\/](?!react-sticky-box|rc-tween-one|@babel|@ant-design|countup.js|_react-sticky-box|_rc-tween-one|_@babel|_@ant-design|_countup.js|\.pnpm)[^\\/]+?[\\/](?!(es)[\\/])```
测试用例：
```
/node_modules/rc-resize-observer/es/
/node_modules/rc-resize-observer/lib/
```

Comment (yoyo837) [reply .jest.js:22]: 仔细看了一下中括号里的, 感觉是不是运行时基于node:path得到分隔符会不会好一点?

Comment (KotoriK) [reply .jest.js:22]: [\\/]是一个社区的通行做法，感觉应该会更为易读

Comment (yoyo837) [inline .jest.js:22]:
```suggestion
  `[/\\]node_modules[/\\](?!${ignoreList.join('|')})[^/\\]+?[/\\](?!(es)[/\\])`,
```

File: .jest.js [hunk 2]
```diff
@@ -12,7 +19,7 @@ const ignoreList = [];
 const transformIgnorePatterns = [
   // Ignore modules without es dir.
   // Update: @babel/runtime should also be transformed
-  `/node_modules/(?!${ignoreList.join('|')})[^/]+?/(?!(es)/)`,
+  `[/\\]node_modules[/\\](?!${ignoreList.join('|')})[^/\\]+?[/\\](?!(es)[/\\])`,
```
Comment (yoyo837) [inline .jest.js:22]:
```suggestion
  `[/\\\\]node_modules[/\\\\](?!${ignoreList.join('|')})[^/\\\\]+?[/\\\\](?!(es)[/\\\\])`,
```

Comment (KotoriK) [reply .jest.js:22]: 这样你确认不会炸嘛 看看你的case

Comment (yoyo837) [reply .jest.js:22]: 就是因为炸了我才重新suggestion的

Comment (yoyo837) [reply .jest.js:22]: 这个表达式在字符串里面, 需要双重转义

Comment (KotoriK) [reply .jest.js:22]: 那么为什么不采用原本的方案呢

Comment (yoyo837) [reply .jest.js:22]: 你原本的方案虽然case通过, 但是实际是不起作用的:

image

Comment (KotoriK) [reply .jest.js]: ok

**Review by afc163 — APPROVED** [review]

**Pull request merged**